astropy/coordinates/builtin_frames/ecliptic_transforms.py

```diff
@@ -105,15 +105,14 @@ def icrs_to_helioecliptic(from_coo, to_frame):
     if not u.m.is_equivalent(from_coo.cartesian.x.unit):
         raise UnitsError(_NEED_ORIGIN_HINT.format(from_coo.__class__.__name__))
 
-    # get barycentric sun coordinate
-    # this goes here to avoid circular import errors
-    from astropy.coordinates.solar_system import get_body_barycentric
-    bary_sun_pos = get_body_barycentric('sun', to_frame.obstime)
+    # get the offset of the barycenter from the Sun
+    sun_bary_pos = offset_sun_from_barycenter(to_frame.obstime, reverse=True,
```

Comment:
Here and below, I find the name rather confusion. Maybe `sun_to_bary_offset`? Alternatively, maybe it is an idea to just put the function call in the return statement?

Reply:
- For the ICRS<->HCRS transformations, I put the function call in the return statement.
- For the ecliptic transformations, I changed the names `sun_from_ssb` and `ssb_from_sun`.  Putting it in the return statement would get too unreadable since half of the transformations require a rotation of the offset.

Reply:
Nice!